tests/test_generator.py

```diff
@@ -33,3 +34,24 @@ def test_boolean():
     }
     output = generate_json(schema)
     assert isinstance(output, bool)
+
+
+def test_pattern_string():
+    """Test generating a pattern restricted string."""
+    # (test uses a pattern for a simple North American telephone number with an optional area code)
+    schema = {
+        "type": "string",
+        "pattern": "^(\\([0-9]{3}\\))?[0-9]{3}-[0-9]{4}$"
+    }
+    output = generate_json(schema)
+    assert isinstance(output, str)
+    assert re.fullmatch(schema.get("pattern"), output) != None
```

Comment:
You should use `is None` and `is not None` as a general rule: <https://stackoverflow.com/questions/3257919/what-is-the-difference-between-is-none-and-none>